protocol-specs/generators/dpos_random_seed_generation/index.js

```diff
@@ -0,0 +1,497 @@
+		),
+		...seedRevealsForRandomSeed1,
+	]);
+	const randomSeed2 = bitwiseXOR([
+		strippedHash(numberToBuffer(firstBlockHeightOfCurrentRound - 1)),
+		...seedRevealsForRandomSeed2,
+	]);
+
+	return {
+		randomSeed1: randomSeed1.toString('hex'),
+		randomSeed2: randomSeed2.toString('hex'),
+	};
+};
+
+const randomSeedFirstRound = () => ({
+	title: 'Random seed for first round',
+	summary: 'Random seeds generation for first round',
+	config: { network: 'devnet' },
+	runner: 'dpos_random_seed_generation',
+	handler: 'dpos_random_seed_generation_first_round',
+	testCases: (() => {
+		const blocksPerRound = activeDelegates + standByDelegates;
+		const blocks = generateBlocks({
+			startHeight: 1,
+			numberOfBlocks: blocksPerRound,
+			delegateList: sampleDelegateList.slice(0, blocksPerRound),
+		});
+		const { randomSeed1, randomSeed2 } = generateRandomSeed(
+			blocks,
+			blocksPerRound,
+		);
+
+		return [
+			{
+				description: 'Random seeds generation for first round',
+				config: { blocksPerRound },
+				input: {
+					blocks,
+				},
+				output: {
+					randomSeed1,
+					randomSeed2,
+				},
+			},
+		];
+	})(),
+});
+
+const randomSeedForMoreRounds = () => ({
+	title: 'Random seed for more than one rounds',
+	summary: 'Random seeds generation more than one rounds',
+	config: { network: 'devnet' },
+	runner: 'dpos_random_seed_generation',
+	handler: 'dpos_random_seed_generation_other_rounds',
+	testCases: (() => {
+		const blocksPerRound = activeDelegates + standByDelegates;
+		const blocksForTwoRounds = generateBlocks({
+			startHeight: 1,
+			numberOfBlocks: blocksPerRound * 2,
+			delegateList: sampleDelegateList.slice(0, blocksPerRound),
+		});
+		const blocksForThreeRounds = generateBlocks({
+			startHeight: 1,
+			numberOfBlocks: blocksPerRound * 3,
+			delegateList: sampleDelegateList.slice(0, blocksPerRound),
+		});
+
+		return [
+			{
+				description: 'Random seeds generation for two rounds',
+				input: {
+					blocksPerRound,
+					blocks: blocksForTwoRounds,
+				},
+				output: generateRandomSeed(blocksForTwoRounds, blocksPerRound),
+			},
+			{
+				description: 'Random seeds generation for three rounds',
+				config: { blocksPerRound },
+				input: {
+					blocks: blocksForTwoRounds,
+				},
+				output: generateRandomSeed(blocksForThreeRounds, blocksPerRound),
+			},
+		];
+	})(),
+});
+
+const randomSeedIfNotPassedMiddleOfRound = () => ({
+	title: 'Random seed for round not passed the middle of the round',
+	summary: 'Random seed for round not passed the middle of the round',
+	config: { network: 'devnet' },
+	runner: 'dpos_random_seed_generation',
+	handler: 'dpos_random_seed_generation_not_passed_middle_of_round',
+	testCases: (() => {
+		const blocksPerRound = activeDelegates + standByDelegates;
+		const blocks = generateBlocks({
+			startHeight: 1,
+			numberOfBlocks: blocksPerRound + 2,
+			delegateList: sampleDelegateList.slice(0, blocksPerRound),
+		});
+
+		const randomSeed1 = null;
```

Comment:
can we use `undefined` here? for the tslint rule, we have `no-null-keyword`

Reply:
I first did `undefined` but it is never get exported to JSON file.

Reply:
ah that's true then it's fine